lisa/base_tools/service.py

```diff
@@ -58,6 +68,95 @@ def is_service_inactive(self, name: str) -> bool:
     def is_service_running(self, name: str) -> bool:
         return self._internal_tool._check_service_running(name)  # type: ignore
 
+    def wait_for_service_start(self, name: str) -> None:
+        raise NotImplementedError()
+
+
+class WindowsService(Tool):
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        self._command = ""
+        self._powershell = self.node.tools[PowerShell]
+
+    @property
+    def can_install(self) -> bool:
+        return False
+
+    @property
+    def command(self) -> str:
+        return self._command
+
+    def _check_exists(self) -> bool:
+        return True
+
+    def restart_service(self, name: str, ignore_exit_code: int = 0) -> None:
+        self._powershell.run_cmdlet(
+            f"Restart-service {name}",
+            force_run=True,
+        )
+        self.wait_for_service_start(name)
+
+    def wait_for_service_start(self, name: str) -> None:
+        for _ in range(10):
+            service_status = self._powershell.run_cmdlet(
+                f"Get-Service {name}",
+                force_run=True,
+                output_json=True,
+                fail_on_error=False,
+            )
+            if schema.WindowsServiceStatus.RUNNING == schema.WindowsServiceStatus(
+                service_status["Status"]
+            ):
+                return
+
+            self._log.debug(
+                f"service '{name}' is not ready yet, retrying... after 3 seconds"
+            )
+            sleep(3)
+
+        raise LisaException(f"service '{name}' failed to start")
+
+    def _get_status(self, name: str = "") -> schema.WindowsServiceStatus:
+        service_status = self._powershell.run_cmdlet(
+            f"Get-Service {name}",
+            force_run=True,
+            output_json=True,
+            fail_on_error=False,
+        )
+        if not service_status:
+            raise LisaException(f"service '{name}' does not exist")
+        return schema.WindowsServiceStatus(service_status["Status"])
+
+    def stop_service(self, name: str) -> None:
+        self._powershell.run_cmdlet(
+            f"Stop-Service {name} -Force",
+            force_run=True,
+            output_json=True,
+            fail_on_error=False,
+        )
+        assert_that(self._get_status(name)).described_as(
+            f"Failed to stop service {name}"
+        ).is_not_equal_to(schema.WindowsServiceStatus.RUNNING)
+
+    def enable_service(self, name: str) -> None:
+        raise NotImplementedError()
+
+    def check_service_status(self, name: str) -> bool:
```

Comment:
Implement the internal `_check_service_running`, instead of replacing the public method.